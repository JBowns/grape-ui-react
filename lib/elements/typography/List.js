'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.List = undefined;

var _templateObject = _taggedTemplateLiteral(['\n    ', '\n    margin: 0 0 ', ';\n    padding-left:  ', ';\n    ', '\n    ', '\n  '], ['\n    ', '\n    margin: 0 0 ', ';\n    padding-left:  ', ';\n    ', '\n    ', '\n  ']);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _styledComponents = require('styled-components');

var _styledComponents2 = _interopRequireDefault(_styledComponents);

var _textStyles = require('./textStyles');

var _globalStyles = require('../../global-styles');

var _utils = require('./utils');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _taggedTemplateLiteral(strings, raw) { return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

var _getGlobalStyles = (0, _globalStyles.getGlobalStyles)(),
    gridSchema = _getGlobalStyles.grid;

var getInLineStyles = function getInLineStyles(inline) {
  var inlineStyles = '\n    > li {\n      display: inline-block;\n      &:not(:last-child) {\n        margin-right: ' + (0, _utils.getScaledSize)(gridSchema.gutter, 0.5) + ';\n      }\n    }\n  ';
  return '' + (inline ? inlineStyles : '');
};

var listFactory = function listFactory(props) {
  var inline = props.inline,
      unstyled = props.unstyled,
      listTag = props.tag,
      otherProps = _objectWithoutProperties(props, ['inline', 'unstyled', 'tag']);

  var tag = listTag || 'ul';

  var ProtoList = _styledComponents2.default[tag](_templateObject, (0, _textStyles.textStylesBase)(), gridSchema.gutter, unstyled || inline ? '0' : '2.5rem', unstyled ? '> li { list-style: none; }' : '', getInLineStyles(inline));
  return _react2.default.createElement(ProtoList, otherProps);
};

listFactory.propTypes = {
  inline: _propTypes2.default.bool,
  tag: _propTypes2.default.string,
  unstyled: _propTypes2.default.bool
};

listFactory.defaultProps = {
  inline: false,
  tag: 'ul',
  unstyled: false
};

var List = function List(props) {
  return listFactory(props);
};

List.ul = List;
List.ol = function (props) {
  return listFactory(Object.assign({}, props, { tag: 'ol' }));
};

exports.List = List;