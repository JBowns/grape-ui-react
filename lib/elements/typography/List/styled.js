import _Object$defineProperty from "@babel/runtime-corejs3/core-js-stable/object/define-property";
import _Object$defineProperties from "@babel/runtime-corejs3/core-js-stable/object/define-properties";
import _Object$getOwnPropertyDescriptors from "@babel/runtime-corejs3/core-js-stable/object/get-own-property-descriptors";
import _forEachInstanceProperty from "@babel/runtime-corejs3/core-js-stable/instance/for-each";
import _Object$getOwnPropertyDescriptor from "@babel/runtime-corejs3/core-js-stable/object/get-own-property-descriptor";
import _filterInstanceProperty from "@babel/runtime-corejs3/core-js-stable/instance/filter";
import _Object$getOwnPropertySymbols from "@babel/runtime-corejs3/core-js-stable/object/get-own-property-symbols";
import _taggedTemplateLiteral from "@babel/runtime-corejs3/helpers/taggedTemplateLiteral";
import _Object$keys from "@babel/runtime-corejs3/core-js-stable/object/keys";
import _objectWithoutProperties from "@babel/runtime-corejs3/helpers/objectWithoutProperties";
import _defineProperty from "@babel/runtime-corejs3/helpers/defineProperty";

function _templateObject() {
  var data = _taggedTemplateLiteral(["\n    ", "\n    ", "\n    ", "\n    ", "\n    ", "\n    ", "\n    ", "\n    ", "\n    ", "\n    ", "\n    ", "\n    ", "\n    ", "\n    ", "\n    ", "\n  "]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function ownKeys(object, enumerableOnly) { var keys = _Object$keys(object); if (_Object$getOwnPropertySymbols) { var symbols = _Object$getOwnPropertySymbols(object); if (enumerableOnly) symbols = _filterInstanceProperty(symbols).call(symbols, function (sym) { return _Object$getOwnPropertyDescriptor(object, sym).enumerable; }); keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { var _context; _forEachInstanceProperty(_context = ownKeys(Object(source), true)).call(_context, function (key) { _defineProperty(target, key, source[key]); }); } else if (_Object$getOwnPropertyDescriptors) { _Object$defineProperties(target, _Object$getOwnPropertyDescriptors(source)); } else { var _context2; _forEachInstanceProperty(_context2 = ownKeys(Object(source))).call(_context2, function (key) { _Object$defineProperty(target, key, _Object$getOwnPropertyDescriptor(source, key)); }); } } return target; }

import React from 'react';
import PropTypes from 'prop-types';
import propTypes from '@styled-system/prop-types';
import styled from 'styled-components';
import { getGlobalStyles } from 'src/global-styles';
import { fontWeight, space } from 'styled-system';
import { colorCore, ellipsisCore, defaultStylesBase, fontFamilyCore, fontSizeCore, fontStyleCore, letterSpacingCore, lineHeightCore, scaleFont, textAlignCore, textDecorationCore, typography } from 'src/utils/styledHelpers';
import { removeSomeProps } from 'src/utils/componentHelpers';

var _getGlobalStyles = getGlobalStyles(),
    gridSchema = _getGlobalStyles.grid;

var inlineStyle = function inlineStyle(props) {
  var style = "\n    > li {\n      display: inline-block;\n      &:not(:last-child) {\n        margin-right: ".concat(scaleFont(gridSchema.gutter, 0.5), ";\n      }\n    }\n  ");
  return "".concat(props.inline ? style : '');
};

var paddingLeft = function paddingLeft(props) {
  return "padding-left:  ".concat(props.unstyled || props.inline ? '0' : '2.5rem', ";");
};

var listStyle = function listStyle(props) {
  return "".concat(props.unstyled ? '> li { list-style: none; }' : '');
};

var margin = function margin(props) {
  return "margin: 0 0 ".concat(gridSchema.gutter, ";");
};

var listComponentPropsToTrim = _objectSpread({}, propTypes.space, {}, typography.propTypes, {
  inline: PropTypes.bool,
  unstyled: PropTypes.bool
});

var listFactory = function listFactory(factoryProps) {
  var tag = factoryProps.tag;

  var ListComponent = function ListComponent(_ref) {
    var children = _ref.children,
        props = _objectWithoutProperties(_ref, ["children"]);

    return React.createElement(tag, removeSomeProps(props, _Object$keys(listComponentPropsToTrim)), children);
  };

  ListComponent.propTypes = {
    children: PropTypes.any.isRequired
  };
  return styled(ListComponent)(_templateObject(), colorCore, ellipsisCore, fontFamilyCore, fontSizeCore, fontWeight, letterSpacingCore, lineHeightCore, fontStyleCore, textAlignCore, textDecorationCore, margin, paddingLeft, listStyle, inlineStyle, space);
};

var List = listFactory({
  tag: 'ul'
});
List.ul = List;
List.ol = listFactory({
  tag: 'ol'
});
List.propTypes = _objectSpread({}, typography.propTypes, {
  inline: PropTypes.bool,
  unstyled: PropTypes.bool
});
List.defaultProps = _objectSpread({}, defaultStylesBase);
/** @component */

export { List };