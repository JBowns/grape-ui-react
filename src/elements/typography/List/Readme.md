Use `<List>` and `<List.ul>` for an unordered list.

```jsx in Markdown
import { ListItem } from '../';

<List>
  <ListItem>
    List Item 1
  </ListItem>
  <ListItem>
    List Item 2
    <List.ul>
      <ListItem>
        List Item 1
      </ListItem>
      <ListItem>
        List Item 2
      </ListItem>
      <ListItem>
        List Item 3
      </ListItem>
    </List.ul>
  </ListItem>
  <ListItem>
    List Item 3
  </ListItem>
</List>
```

Use `<List.ol>` instead of `<List>` for an ordered list.

```jsx in Markdown
import { ListItem } from '../';

<List.ol>
  <ListItem>
    List Item 1
  </ListItem>
  <ListItem>
    List Item 2
  </ListItem>
  <ListItem>
    List Item 3
  </ListItem>
</List.ol>
```

`<List.ol>` takes the standard `type` attribute to define if uppercase alpha, lowercase alpha, uppercase roman, lowercase roman.

```jsx in Markdown
import { Code, ListItem } from 'src/elements/typography';

<div>
  <List.ol type="A">
    <ListItem>
      {'Uppercase Alpha List Item 1 '}
      <Code codeString={'type="A"'} />
    </ListItem>
    <ListItem>
      {'Uppercase Alpha List Item 2 '}
      <Code codeString={'type="A"'} />
    </ListItem>
    <ListItem>
      {'Uppercase Alpha List Item 3 '}
      <Code codeString={'type="A"'} />
    </ListItem>
  </List.ol>
  <List.ol type="a">
    <ListItem>
      {'Lowercase Alpha List Item 1 '}
      <Code codeString={'type="a"'} />
    </ListItem>
    <ListItem>
      {'Lowercase Alpha List Item 2 '}
      <Code codeString={'type="a"'} />
    </ListItem>
    <ListItem>
      {'Lowercase Alpha List Item 3 '}
      <Code codeString={'type="a"'} />
    </ListItem>
  </List.ol>
  <List.ol type="I">
    <ListItem>
      {'Uppercase Roman List Item 1 '}
      <Code codeString={'type="I"'} />
    </ListItem>
    <ListItem>
      {'Uppercase Roman List Item 2 '}
      <Code codeString={'type="I"'} />
    </ListItem>
    <ListItem>
      {'Uppercase Roman List Item 3 '}
      <Code codeString={'type="I"'} />
    </ListItem>
  </List.ol>
  <List.ol type="i">
    <ListItem>
      {'Lowercase Roman List Item 1 '}
      <Code codeString={'type="i"'} />
    </ListItem>
    <ListItem>
      {'Lowercase Roman List Item 2 '}
      <Code codeString={'type="i"'} />
    </ListItem>
    <ListItem>
      {'Lowercase Roman List Item 3 '}
      <Code codeString={'type="i"'} />
    </ListItem>
  </List.ol>
</div>
```

### Style variations

Add the attribute `unstyled` to have an unstyled view. Add the attribute `inline` to have an inline view.

```jsx in Markdown
import { Code, ListItem } from 'src/elements/typography';

<div>
  <List unstyled>
    <ListItem>
      {'This List component was set to '}
      <Code codeString="unstyled" />
      .
    </ListItem>
    <ListItem>
      List Item 2
    </ListItem>
    <ListItem>
      List Item 3
    </ListItem>
  </List>
  <List inline>
    <ListItem>
      {'This List component was set to '}
      <Code codeString="inline" />
      .
    </ListItem>
    <ListItem>
      List Item 2
    </ListItem>
    <ListItem>
      List Item 3
    </ListItem>
  </List>
</div>
```
