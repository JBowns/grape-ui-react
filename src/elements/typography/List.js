import PropTypes from 'prop-types';
import styled from 'styled-components';
import { getGlobalStyles } from 'src/global-styles';
import { getScaledSize } from './utils';
import {
  getFontFamily,
  getFontSize,
  getFontWeight,
  getLetterSpacing,
  getLineHeight,
  getFontStyle,
  getColor,
  getTextAlign,
  getTextDecoration,
} from './textStyles';

const { grid: gridSchema } = getGlobalStyles();

const getInLineStyles = props => {
  const inlineStyles = `
    > li {
      display: inline-block;
      &:not(:last-child) {
        margin-right: ${getScaledSize(gridSchema.gutter, 0.5)};
      }
    }
  `;
  return `${props.inline ? inlineStyles : ''}`;
};

const getPaddingLeft = props => `padding-left:  ${(props.unstyled || props.inline) ? '0' : '2.5rem'};`;
const getListStyle = props => `${props.unstyled ? '> li { list-style: none; }' : ''}`;

const ListFactory = factoryProps => {
  const { tag } = factoryProps;
  return styled[tag]`
    ${getFontFamily}
    ${getFontSize}
    ${getFontWeight}
    ${getLetterSpacing}
    ${getLineHeight}
    ${getFontStyle}
    ${getColor}
    ${getTextAlign}
    ${getTextDecoration}
    margin: 0 0 ${gridSchema.gutter};
    ${getPaddingLeft}
    ${getListStyle}
    ${getInLineStyles}
    `;
};

const List = ListFactory({ tag: 'ul' });

List.ul = List;
List.ol = ListFactory({ tag: 'ol' });
List.propTypes = {
  color: PropTypes.string,
  display: PropTypes.bool,
  fontFamily: PropTypes.string,
  fontWeight: PropTypes.string,
  inline: PropTypes.bool,
  kerning: PropTypes.string,
  lg: PropTypes.bool,
  sm: PropTypes.bool,
  textAlign: PropTypes.string,
  textDecoration: PropTypes.string,
  unstyled: PropTypes.bool,
};
export { List };
