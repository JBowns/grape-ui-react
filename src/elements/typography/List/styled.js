import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { getGlobalStyles } from 'src/global-styles';
import { getScaledSize } from '../utils';
import {
  getFontFamily,
  getFontSize,
  getFontWeight,
  getLetterSpacing,
  getLineHeight,
  getFontStyle,
  getColor,
  getTextAlign,
  getTextDecoration,
  typography,
} from '../textStyles';
import { passThrough } from 'src/utils/componentHelpers';

const { grid: gridSchema } = getGlobalStyles();

const getInLineStyles = props => {
  const inlineStyles = `
    > li {
      display: inline-block;
      &:not(:last-child) {
        margin-right: ${getScaledSize(gridSchema.gutter, 0.5)};
      }
    }
  `;
  return `${props.inline ? inlineStyles : ''}`;
};

const getPaddingLeft = props => `padding-left:  ${(props.unstyled || props.inline) ? '0' : '2.5rem'};`;
const getListStyle = props => `${props.unstyled ? '> li { list-style: none; }' : ''}`;

const listFactory = factoryProps => {
  const { tag } = factoryProps;
  const ListComponent = ({
    children, ...props
  }) => React.createElement(tag, passThrough(ListComponent, props), children);
  ListComponent.propTypes = {
    children: PropTypes.any.isRequired,
    inline: PropTypes.bool,
    ...typography.propTypes,
    unstyled: PropTypes.bool,
  };
  ListComponent.defaultProps = {
    inline: false,
    unstyled: false,
  };

  return styled(ListComponent)`
    ${getFontFamily}
    ${getFontSize}
    ${getFontWeight}
    ${getLetterSpacing}
    ${getLineHeight}
    ${getFontStyle}
    ${getColor}
    ${getTextAlign}
    ${getTextDecoration}
    margin: 0 0 ${gridSchema.gutter};
    ${getPaddingLeft}
    ${getListStyle}
    ${getInLineStyles}
    `;
};

const List = listFactory({ tag: 'ul' });
List.ul = List;
List.ol = listFactory({ tag: 'ol' });
List.propTypes = {
  inline: PropTypes.bool,
  ...typography.propTypes,
  unstyled: PropTypes.bool,
};
export { List };
