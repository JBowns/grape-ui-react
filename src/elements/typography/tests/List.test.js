import React from 'react';
import { List } from '../List';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

describe('List Component', () => {
  it('should return a List object with font-style: base', () => {
    const component = renderer.create(<List>Lorem Ipsum</List>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should return a List object with font-style: italic', () => {
    const component = renderer.create(<List italic>Lorem Ipsum</List>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should return a List object with font-size: lg', () => {
    const component = renderer.create(<List lg>Lorem Ipsum</List>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should return a List object with font-size: sm', () => {
    const component = renderer.create(<List sm>Lorem Ipsum</List>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('List Specific Styling', () => {
  it('should return an unstyled List object', () => {
    const component = renderer.create(<List unstyled>Lorem Ipsum</List>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should return an inline List object', () => {
    const component = renderer.create(<List inline>Lorem Ipsum</List>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Numbered List Component', () => {
  it('should return a Numbered List object with font-style: base', () => {
    const component = renderer.create(<List.ol>Lorem Ipsum</List.ol>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should return a Numbered List object with font-style: italic', () => {
    const component = renderer.create(<List.ol italic>Lorem Ipsum</List.ol>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should return a Numbered Numbered List object with font-size: lg', () => {
    const component = renderer.create(<List.ol lg>Lorem Ipsum</List.ol>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should return a Numbered List object with font-size: sm', () => {
    const component = renderer.create(<List.ol sm>Lorem Ipsum</List.ol>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Numbered List Specific Styling', () => {
  it('should return an unstyled Numbered List object', () => {
    const component = renderer.create(<List.ol unstyled>Lorem Ipsum</List.ol>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should return an inline Numbered List object', () => {
    const component = renderer.create(<List.ol inline>Lorem Ipsum</List.ol>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
