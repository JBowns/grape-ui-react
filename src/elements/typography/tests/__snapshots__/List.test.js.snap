// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Component should return a List object with base styling 1`] = `
.c0 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: calc(1rem * 1);
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  color: inherit;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  margin: 0 0 1rem;
  padding-left: 2.5rem;
}

<ul
  className="c0"
>
  Lorem Ipsum
</ul>
`;

exports[`List Component should return a List object with font-size: lg 1`] = `
.c0 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: calc(1rem * 1);
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  color: inherit;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  margin: 0 0 1rem;
  padding-left: 2.5rem;
}

<ul
  className="c0"
>
  Lorem Ipsum
</ul>
`;

exports[`List Component should return a List object with font-size: sm 1`] = `
.c0 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: calc(1rem * 1);
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  color: inherit;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  margin: 0 0 1rem;
  padding-left: 2.5rem;
}

<ul
  className="c0"
>
  Lorem Ipsum
</ul>
`;

exports[`List Component should return a List object with font-style: italic 1`] = `
.c0 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: calc(1rem * 1);
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  color: inherit;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  margin: 0 0 1rem;
  padding-left: 2.5rem;
}

<ul
  className="c0"
>
  Lorem Ipsum
</ul>
`;

exports[`List Specific Styling should return an inline List object 1`] = `
.c0 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: calc(1rem * 1);
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  color: inherit;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  margin: 0 0 1rem;
  padding-left: 0;
}

.c0 > li {
  display: inline-block;
}

.c0 > li:not(:last-child) {
  margin-right: calc(1rem / 2);
}

<ul
  className="c0"
>
  Lorem Ipsum
</ul>
`;

exports[`List Specific Styling should return an unstyled List object 1`] = `
.c0 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: calc(1rem * 1);
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  color: inherit;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  margin: 0 0 1rem;
  padding-left: 0;
}

.c0 > li {
  list-style: none;
}

<ul
  className="c0"
>
  Lorem Ipsum
</ul>
`;

exports[`Numbered List Component should return a List object with Alpha lowercase 1`] = `
.c0 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: calc(1rem * 1);
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  color: inherit;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  margin: 0 0 1rem;
  padding-left: 2.5rem;
}

<ol
  className="c0"
  type="a"
>
  Lorem Ipsum
</ol>
`;

exports[`Numbered List Component should return a List object with Alpha uppercase 1`] = `
.c0 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: calc(1rem * 1);
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  color: inherit;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  margin: 0 0 1rem;
  padding-left: 2.5rem;
}

<ol
  className="c0"
  type="A"
>
  Lorem Ipsum
</ol>
`;

exports[`Numbered List Component should return a Numbered List object with base styling 1`] = `
.c0 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: calc(1rem * 1);
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  color: inherit;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  margin: 0 0 1rem;
  padding-left: 2.5rem;
}

<ol
  className="c0"
>
  Lorem Ipsum
</ol>
`;

exports[`Numbered List Component should return a Numbered List object with font-size: sm 1`] = `
.c0 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: calc(1rem * 1);
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  color: inherit;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  margin: 0 0 1rem;
  padding-left: 2.5rem;
}

<ol
  className="c0"
>
  Lorem Ipsum
</ol>
`;

exports[`Numbered List Component should return a Numbered List object with font-style: italic 1`] = `
.c0 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: calc(1rem * 1);
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  color: inherit;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  margin: 0 0 1rem;
  padding-left: 2.5rem;
}

<ol
  className="c0"
>
  Lorem Ipsum
</ol>
`;

exports[`Numbered List Component should return a Numbered List object with roman numerals lowercase 1`] = `
.c0 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: calc(1rem * 1);
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  color: inherit;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  margin: 0 0 1rem;
  padding-left: 2.5rem;
}

<ol
  className="c0"
  type="i"
>
  Lorem Ipsum
</ol>
`;

exports[`Numbered List Component should return a Numbered List object with roman numerals uppercase 1`] = `
.c0 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: calc(1rem * 1);
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  color: inherit;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  margin: 0 0 1rem;
  padding-left: 2.5rem;
}

<ol
  className="c0"
  type="I"
>
  Lorem Ipsum
</ol>
`;

exports[`Numbered List Component should return a Numbered Numbered List object with font-size: lg 1`] = `
.c0 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: calc(1rem * 1);
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  color: inherit;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  margin: 0 0 1rem;
  padding-left: 2.5rem;
}

<ol
  className="c0"
>
  Lorem Ipsum
</ol>
`;

exports[`Numbered List Specific Styling should return an inline Numbered List object 1`] = `
.c0 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: calc(1rem * 1);
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  color: inherit;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  margin: 0 0 1rem;
  padding-left: 0;
}

.c0 > li {
  display: inline-block;
}

.c0 > li:not(:last-child) {
  margin-right: calc(1rem / 2);
}

<ol
  className="c0"
>
  Lorem Ipsum
</ol>
`;

exports[`Numbered List Specific Styling should return an unstyled Numbered List object 1`] = `
.c0 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: calc(1rem * 1);
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  color: inherit;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  margin: 0 0 1rem;
  padding-left: 0;
}

.c0 > li {
  list-style: none;
}

<ol
  className="c0"
>
  Lorem Ipsum
</ol>
`;
